acme/acme/client.py

```diff
@@ -245,365 +234,6 @@ def _revoke(self, cert: jose.ComparableX509, rsn: int, url: str) -> None:
-        :type certr: `.CertificateResource`
-
-        :returns: Updated Certificate Resource.
-        :rtype: `.CertificateResource`
-
-        """
-        # TODO: acme-spec 5.1 table action should be renamed to
-        # "refresh cert", and this method integrated with self.refresh
-        response, cert = self._get_cert(certr.uri)
-        if 'Location' not in response.headers:
-            raise errors.ClientError('Location header missing')
-        if response.headers['Location'] != certr.uri:
-            raise errors.UnexpectedUpdate(response.text)
-        return certr.update(body=cert)
-
-    def refresh(self, certr: messages.CertificateResource) -> messages.CertificateResource:
-        """Refresh certificate.
-
-        :param certr: Certificate Resource
-        :type certr: `.CertificateResource`
-
-        :returns: Updated Certificate Resource.
-        :rtype: `.CertificateResource`
-
-        """
-        # TODO: If a client sends a refresh request and the server is
-        # not willing to refresh the certificate, the server MUST
-        # respond with status code 403 (Forbidden)
-        return self.check_cert(certr)
-
-    def fetch_chain(self, certr: messages.CertificateResource,
-                    max_length: int = 10) -> List[jose.ComparableX509]:
-        """Fetch chain for certificate.
-
-        :param .CertificateResource certr: Certificate Resource
-        :param int max_length: Maximum allowed length of the chain.
-            Note that each element in the certificate requires new
-            ``HTTP GET`` request, and the length of the chain is
-            controlled by the ACME CA.
-
-        :raises errors.Error: if recursion exceeds `max_length`
-
-        :returns: Certificate chain for the Certificate Resource. It is
-            a list ordered so that the first element is a signer of the
-            certificate from Certificate Resource. Will be empty if
-            ``cert_chain_uri`` is ``None``.
-        :rtype: `list` of `OpenSSL.crypto.X509` wrapped in `.ComparableX509`
-
-        """
-        chain: List[jose.ComparableX509] = []
-        uri = certr.cert_chain_uri
-        while uri is not None and len(chain) < max_length:
-            response, cert = self._get_cert(uri)
-            uri = response.links.get('up', {}).get('url')
-            chain.append(cert)
-        if uri is not None:
-            raise errors.Error(
-                "Recursion limit reached. Didn't get {0}".format(uri))
-        return chain
-
-    def revoke(self, cert: jose.ComparableX509, rsn: int) -> None:
-        """Revoke certificate.
-
-        :param .ComparableX509 cert: `OpenSSL.crypto.X509` wrapped in
-            `.ComparableX509`
-
-        :param int rsn: Reason code for certificate revocation.
-
-        :raises .ClientError: If revocation is unsuccessful.
-
-        """
-        self._revoke(cert, rsn, self.directory[messages.Revocation])
-
-
 class ClientV2(ClientBase):
```

Comment:
Ah, just thinking how we also no longer technically need `ClientBase`, and I doubt anyone was using it directly. But I guess there's no harm to keeping it really.

Reply:
Let me know if you'd like me to revert or partially revert the latest commit I've made, because it's a bit of a doozy. You made me realize there was still quite a bit of ACMEv1 code hanging around and I've made an attempt to remove it. 

It has ended up with more public API removals than we had announced so far. Tbh it was probably inevitable that we'd find more things to remove.

Reply:
Agreed about it being inevitable. Going to take the time to make sure everything's good since it was such a big change, but in general I'm glad to see things simplified, even if it means changing more. That being said, if we're going to remove more things, we should stick in a warning about those ASAP.

Reply:
I have created #9365 to track new things we want to deprecate in the next release, including `ClientBase`.